src/java/com/unifina/signalpath/ModuleException.java

```diff
@@ -13,4 +16,29 @@ public ModuleException(String message, Throwable cause, List<ModuleExceptionMess
 	public List<ModuleExceptionMessage> getModuleExceptions() {
 		return moduleExceptions;
 	}
+
+	public static class CompileError extends HashMap<String, Object> {
+		public CompileError(Long line, String message) {
+			put("line", line);
+			put("message", message);
+		}
+	}
+	public List<CompileError> getCompilerErrors() {
+		final List<CompileError> errors = new ArrayList<>();
+		for (ModuleExceptionMessage msg : getModuleExceptions()) {
+			for (Map map : msg.getErrors()) {
+				Long line;
+				try {
+					line = Long.parseLong(map.get("line").toString());
+				} catch (NumberFormatException e) {
+					line = new Long(0);
+				}
+				final String message = map.get("msg").toString();
+				final CompileError error = new CompileError(line, message);
```

Comment:
- All `ModuleException`s may not be compile errors! They are module specific! It would be better if the `ModuleExceptionMessage` knew how to render itself, so that the frontend module implementation could do something useful with the information.